src/Microsoft.Extensions.DependencyInjection.Ordered/ServiceCollectionOrderedExtensions.cs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection.Ordered
```

Comment:
If we are going to add a new concept of an Ordered enumerable, should go a bit farther, and at least support an overload that allows insertion @ index?  

How easy is it going to be to look at the current registrations for an `IOrdered<T>`